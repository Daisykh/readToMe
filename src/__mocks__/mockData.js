const mockData = [
  {
    id: 1,
    img: 'https://www.scholastic.com/content5/media/products/24/9780545964524_mres.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 2,
    img: 'http://www.pammunozryan.com/wp-content/uploads/2015/07/Esperana-Rising-Cover.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 3,
    img: 'https://images-na.ssl-images-amazon.com/images/I/71ooSs5M6iL.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 4,
    img: 'https://images-na.ssl-images-amazon.com/images/I/A1vYN3THqlL.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 5,
    img: 'https://fthmb.tqn.com/lboQ3JUKYYOll4zcCzk9cI5G8k8=/768x0/filters:no_upscale()/Wonder-572c22db3df78c038e82ba79.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 6,
    img: 'https://images-na.ssl-images-amazon.com/images/I/81EVdWdmOKL.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 7,
    img: 'https://s3.amazonaws.com/thumbnails.thecrimson.com/photos/2016/03/07/200354_1313657.jpg.800x1179_q95_crop-smart_upscale.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 8,
    img: 'https://images-na.ssl-images-amazon.com/images/I/A15wRHQ3jXL.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 9,
    img: 'https://www.scholastic.com/content5/media/products/55/9780545806855_mres.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 10,
    img: 'https://images-na.ssl-images-amazon.com/images/I/91IkhrdcJ1L.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 11,
    img: 'https://cdn.shopify.com/s/files/1/0384/8265/products/guessiloveu.jpg?v=1403077384',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 12,
    img: 'https://www.scholastic.com/content5/media/products/24/9780545964524_mres.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 13,
    img: 'https://www.scholastic.com/content5/media/products/24/9780545964524_mres.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 14,
    img: 'https://www.scholastic.com/content5/media/products/24/9780545964524_mres.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 15,
    img: 'https://www.scholastic.com/content5/media/products/24/9780545964524_mres.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 16,
    img: 'https://www.scholastic.com/content5/media/products/24/9780545964524_mres.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 17,
    img: 'https://www.scholastic.com/content5/media/products/24/9780545964524_mres.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 18,
    img: 'https://www.scholastic.com/content5/media/products/24/9780545964524_mres.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  },

  {
    id: 19,
    img: 'https://www.scholastic.com/content5/media/products/24/9780545964524_mres.jpg',
    text: 'The quick brown fox jumps over the lazy dog',
  }
]

export default mockData;