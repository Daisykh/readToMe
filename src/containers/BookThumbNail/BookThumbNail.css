a {
  width: 100px;
  height: 200px;
}

#thumb-nail img {
  text-align: center; 
  position: relative;
  z-index: 0;
  height: 200px;
  width: 100px;
  margin: -10px;
}

.thumbNailsContainer {
  width: 95vw;
  display: flex;
  flex-direction: row;
  margin: 0px 8px;
}

.thumbContainer {
  width: 150px;
}

#thumb-nail {
  max-width: 200px;
  padding: 10px;
  overflow: hidden;
  border: 1px solid grey;
}

.row {
  margin: 0;
  width: 70% ;
  height: 325px;
}

.row__inner {
  transition: 450ms transform;
  font-size: 0;
  white-space: nowrap;
  margin: 0;
 /* margin: ((250px / (16/9)) / 2) 0;*/
  padding: 20px;
  height: 290px;
  overflow-y: scroll;
  overflow-x: visible;
 /* display: flex;*/
  /*flex-direction: row;
*/
}

/*come back and make the rows flex*/

.thumb
 { 
  position: relative;
   display: inline-block;
   width: 175px;
   max-height: 160px;
   margin-right: 40px;
   font-size: 20px;
   cursor: pointer;
   transition: 450ms all;
   transform-origin: center;
 }

.thumb__img
{  
  width: 175px;
  height: 100%;
  object-fit: cover;
}

/*.thumb__details:hover {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  font-size: 10px;
  opacity: 0;
  background: linear-gradient(to top, rgba(0,0,0,0.9) 0%, rgba(0,0,0,0) 100%);
  transition: 100ms opacity;
}
*/

/*.thumb__details:before {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  display: black;
}*/

/*.thumb__details:hover {
    margin-top: (-(50px / 2));
    margin-left: (-(50px / 2));
    width: 50px;
    height: 50px;
    border: 3px solid #ecf0f1;
    line-height: 50px;
    text-align: center;
    border-radius: 100%;
    background: rgba(0,0,0,0.5);
    z-index: 1;
  }*/

/*.thumb__details:before {
    content: 'â–¶';
    left: 0;
    width: 100%;
    font-size: 30px;
    margin-left: 7px;
    margin-top: -18px;
    text-align: center;
    z-index: 2;
  }
*/
/*.thumb:hover {
    opacity: 1
}*/

/*.thumb__title
 { 
  position: absolute;
  bottom: 0;
  padding: 10px;
 }*/

.parent:hover > .thumb {
  opacity: 0.5;
}

.parent:hover > .thumb:hover {
  opacity: 2.0;
}

.row__inner:hover
{ 
 transform: translate3d(-(250px * (1.5 - 1) / 2), 0, 0);
/* opacity: .3;
*/}

.row__inner:hover, 
/*.thumb
*/{
 
}

.row-container:hover > .row {
  opacity: 0.8;
}

.row-container:hover > .row:hover {
  opacity: 2.0;
}


.thumb:hover
{  
  transform: scale(1.2);
  opacity: 1;
}

.thumb:hover {
  transform: translate3d(250px * (1.5 - 1), 0, 0);
}
