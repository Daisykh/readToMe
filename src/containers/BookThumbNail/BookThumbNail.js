import React, { Component } from 'react';
import mockData from '../../__mocks__/mockData';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { loadCards } from '../../actions/index';
import { Link, withRouter } from 'react-router-dom';
// import { apiPost } from '../../helper/apiCalls';
import './BookThumbNail.css';
// const recognizeMic = require('watson-speech/speech-to-text/recognize-microphone');

export class BookThumbNail extends Component {

  componentDidMount() {
    const flashCards = mockData;
    // const response = await apiPost();
    // await console.log(response);
    this.props.loadCards(flashCards);
  }

  renderedThumbNails = () => this.props.cards.map( (card, index) => {
    const url = card.img;

    return (
      <div className="thumb" key={index}>
        <Link to={`/cards/${card.id}`} >
          <div className={card.id}>
            <img className="thumb__img" src={url}/>
          </div>
          <div className="thumb__details">
            <div className="thumb__title"></div>
          </div>
        </Link>
      </div>
    );
  })
  

  render() {
    return (
      <div className="row-container">
        <div className="row">
          <div className="row__inner parent">
            { this.renderedThumbNails() }
          </div>
        </div>
        <div className="row">
          <div className="row__inner parent">
            { this.renderedThumbNails() }
          </div>
        </div>
        <div className="row">
          <div className="row__inner parent">
            { this.renderedThumbNails() }
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  cards: state.cards
});

const mapDispatchToProps = (dispatch) => ({
  loadCards: cards => dispatch(loadCards(cards))
});

BookThumbNail.propTypes = {
  loadCards: PropTypes.string,
  cards: PropTypes.string,
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(BookThumbNail));