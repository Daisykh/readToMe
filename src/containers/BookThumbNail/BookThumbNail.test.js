/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import { BookThumbNail, mapStateToProps, mapDispatchToProps, loadCards } from './BookThumbNail';
import mockData from '../../__mocks__/mockData';

describe('BookThumbNail', () => {
  let renderedComponent
  let mockLoadCards
  let mockCards

  // beforeEach(() => {

  // });

  it('should match snapshot', () => {
    const mockLoadCards = jest.fn()
    const mockCards = mockData

    const renderedComponent = 
    shallow(<BookThumbNail 
      cards={ mockData }
      loadCards={ mockLoadCards }
      />, { disableLifecycleMethods: true });
    expect(renderedComponent).toMatchSnapshot();
  })

  it('should map to the store correctly', () => {
    const mockStore = {
      cards: mockData
    }
    const mapped = mapStateToProps(mockStore)

    expect(mapped).toEqual(mockStore)
  });

  it('should call the dispatch function when using a function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mockStore = {
      cards: mockData
    }
    const mapped = mapDispatchToProps(mockDispatch);

    expect(mockDispatch).not.toHaveBeenCalled();

    mapped.loadCards(mockStore);

    expect(mockDispatch).toHaveBeenCalled();
  });
})
