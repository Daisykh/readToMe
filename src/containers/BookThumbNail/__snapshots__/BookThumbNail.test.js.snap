// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookThumbNail should match snapshot 1`] = `undefined`;
